features/step_definitions/icon_steps.rb

```diff
@@ -10,17 +10,35 @@
 Given /^I have an icon uploaded$/ do
   step "I am editing a pseud"
   step %{I attach the file "test/fixtures/icon.gif" to "icon"}
-    step %{I press "Update"}
+  step %{I press "Update"}
 end
 
 ### WHEN
 
-When /^I add an icon to the collection$/ do
+When /^I add an icon to the collection "([^"]*)"$/ do |name|
   step %{I am logged in as "moderator"}
-    step %{I am on "Pretty" collection's page}
-    step %{I follow "Settings"}
-    step %{I attach the file "test/fixtures/icon.gif" to "collection_icon"}
-    step %{I press "Update"}
+  step %{I am on "#{name}" collection's page}
+  step %{I follow "Settings"}
+  step %{I attach the file "test/fixtures/icon.gif" to "collection_icon"}
+  step %{I press "Update"}
+end
+
+When /^I delete the icon from the collection "([^"]*)"$/ do |name|
```

Comment:
Same thing -- I'd suggest changing the name of this variable to `title`, since it's the collection title, not the collection name, that we use for `"Whatever" collection's page`.